File: sdks/sdkwork-mail-sdk/sdkwork-mail-sdk-java/src/main/java/com/sdkwork/Mail/metadata/mailProviderActivationCatalog.java
package com.sdkwork.Mail.metadata;

import java.util.List;
import java.util.Optional;

public final class MailProviderActivationCatalog {

  public static final List<String> RECOGNIZED_ACTIVATION_STATUSES = List.of(
      "package-boundary",
      "control-metadata-only"
  );

  public static final List<MailProviderActivationCatalogEntry> ENTRIES = List.of(
      new MailProviderActivationCatalogEntry("smtp", "Mail-smtp", "sdkwork-mail-driver-smtp", "package-boundary", true, false, true, true, "com.sdkwork:Mail-sdk-provider-smtp"),
      new MailProviderActivationCatalogEntry("imap", "Mail-imap", "sdkwork-mail-driver-imap", "package-boundary", true, false, true, true, "com.sdkwork:Mail-sdk-provider-imap")
  );

public static Optional<MailProviderActivationCatalogEntry> getMailProviderActivationByProviderKey(String providerKey) {
    for (var entry : ENTRIES) {
      if (entry.providerKey().equals(providerKey)) {
        return Optional.of(entry);
      }
    }

    return Optional.empty();
  }


  private MailProviderActivationCatalog() {
  }

  public record MailProviderActivationCatalogEntry(
      String providerKey,
      String pluginId,
      String driverId,
      String activationStatus,
      boolean runtimeBridge,
      boolean rootPublic,
      boolean packageBoundary,
      boolean builtin,
      String packageIdentity
  ) {
  }
}

File: sdks/sdkwork-mail-sdk/sdkwork-mail-sdk-kotlin/src/main/kotlin/com/sdkwork/Mail/metadata/mailProviderActivationCatalog.kt
package com.sdkwork.Mail.metadata

data class MailProviderActivationCatalogEntry(
    val providerKey: String,
    val pluginId: String,
    val driverId: String,
    val activationStatus: String,
    val runtimeBridge: Boolean,
    val rootPublic: Boolean,
    val packageBoundary: Boolean,
    val builtin: Boolean,
    val packageIdentity: String,
)

object MailProviderActivationCatalog {
    val recognizedActivationStatuses: List<String> = listOf(
        "package-boundary",
        "control-metadata-only",
    )

    val entries: List<MailProviderActivationCatalogEntry> = listOf(
        MailProviderActivationCatalogEntry("smtp", "Mail-smtp", "sdkwork-mail-driver-smtp", "package-boundary", true, false, true, true, "com.sdkwork:Mail-sdk-provider-smtp"),
        MailProviderActivationCatalogEntry("imap", "Mail-imap", "sdkwork-mail-driver-imap", "package-boundary", true, false, true, true, "com.sdkwork:Mail-sdk-provider-imap"),
    )

fun getMailProviderActivationByProviderKey(providerKey: String): MailProviderActivationCatalogEntry? =
        entries.firstOrNull { it.providerKey == providerKey }

}
